parse_pwe_templates.py

```diff
@@ -707,6 +708,50 @@ def to_json(self) -> dict:
 
         self.class_code += "        }\n"
 
+    def write_dependencies(self):
+        if self.keep_unknown_properties() or self.game_id not in {"Prime", "Echoes"}:
+            return
+
+        has_dep = False
+
+        for prop_name, prop in self.all_props.items():
+            if prop.dependency_code is None:
+                continue
+
+            has_dep = True
+            prop_code = prop.dependency_code.format(obj=f'self.{prop_name}')
+            if prop.raw.get("ignore_dependencies_mlvl"):
+                prop_code = "if not is_mlvl:\n            " + prop_code
+
+            self.class_code += f"""
+    def _dependencies_for_{prop_name}(self, asset_manager, is_mlvl: bool):
+        {prop_code}
+"""
+
+        if has_dep:
+            method_list = "\n".join(
+                f'            (self._dependencies_for_{prop_name}, "{prop_name}", "{prop.prop_type}"),'
+                for prop_name, prop in self.all_props.items()
+                if prop.dependency_code is not None
+            )
+            self.class_code += f"""
+    def dependencies_for(self, asset_manager, is_mlvl: bool = False):
+        for method, field_name, field_type in [
+{method_list}
+        ]:
+            try:
+                yield from method(asset_manager, is_mlvl)
+            except Exception as e:
+                raise Exception(
+                    f"Error finding dependencies for AreaAttributes.{{field_name}} ({{field_type}}): {{e}}"
```

Comment:
shouldn't this be generic, instead of `AreaAttributes`?

Reply:
OOOPS